### PR TITLE
WT-9935 Add assert to ensure we are not returning overflow removed value to the reader

### PR DESCRIPTION
**Pull request by quchenhao**

### CODE REVIEW

File: src/support/modify.c [hunk 1]
```diff
@@ -484,7 +490,17 @@ __wt_modify_reconstruct_from_upd_list(
...
+        ret = __wt_value_return_buf(cbt, cbt->ref, &upd_value->buf, &tw
+#ifdef HAVE_DIAGNOSTIC
+          ,
+          &is_ovfl_rm
+#endif
+        );
+        WT_ERR(ret);
+
+        /* The base value cannot be a removed overflow value. */
+        WT_ASSERT(session, !is_ovfl_rm);
```
Comment (etienneptl) [inline src/support/modify.c:502]: Should we have it `WT_ASSERT_ALWAYS` for the time being?

Comment (quchenhao) [reply src/support/modify.c:502]: No, this doesn't work in production as we only pass the parameter in debug build.

**Review by etienneptl — APPROVED** [review]

**Review by mm-ng — APPROVED** [review]

**Comment by quchenhao** [comment]
evergreen merge

**Comment by quchenhao** [comment]
evergreen merge

**Pull request merged**